src/MicroPather/micropather.cpp

```diff
@@ -706,7 +705,7 @@ void PathCache::Reset()
 {
 	if (nItems)
 	{
-		memset(mem, 0, sizeof(*mem) * allocated);
+		mem = {};

```

Comment:
This doesn't do what you think it does and does not do the equivalent to what it did before.

`mem` is a dynamic C-array masquerading as a pointer, assigning it via braced initialization assigns it `nullptr`. The previous version zeroed out the memory (`Item` is a POD, so this is safe). This does not.

Conceptually, since `mem` is a dynamic C-array, that's just a `std::vector`

Change `Item* mem;` to `std::vector<Item> mem;`.

The only lines that will need to change by changing the type is 708 and 822 and the c'tor/d'tor:
708 from:
```
mem = {};
```
to
``
//std::vector::clear() does not reduce the capacity, no de-allocation is performed.
mem.clear();
```
and 822 from:
```
return mem + index;
```
to
```
return mem.data() + index;
```
and the constructor/destructor of `PathCache` to:
```
PathCache::PathCache(int _allocated)
{
        mem = std::vector<Item>(_allocated, Item{});
        allocated = _allocated;
        nItems = 0;
        hit = 0;
        miss = 0;
}
```
and the destructor can be `=default`ed or implemented to specifically clear and shrink `mem`:
``
PathCache::~PathCache()
{
        mem.clear();
        mem.shrink_to_fit();
}
```

Reply:
Careless oversight from programming while overly tired. The code should have read:
```cpp
*mem = {};
```

Reply:
Oh, the and type is not a POD by virtue of not being an aggregate type. Hence the warning about it.

Reply:
Just poking the thread. Any update to this?